File: 0202-happy-number/0202-happy-number.c
int sumOfSquares(int number) {
    int rem, sum = 0;

    while (number > 0) {
        rem = number % 10;
        sum += rem * rem;
        number /= 10;
    }

    return sum;
}

bool isHappy(int n) {
    int slow, fast;
    slow = fast = n;

    do {
        slow = sumOfSquares(slow);              
        fast = sumOfSquares(sumOfSquares(fast));
    } while (slow != fast);

    return (slow == 1);
}

File: 0202-happy-number/0202-happy-number.java
class Solution {
    public int sum(int n) {
        int res = 0;
        while (n > 0) {
            int rem = n % 10;
            rem = rem * rem;
            res = res + rem;
            n = n / 10;
        }
        return res;
    }

    public boolean isHappy(int n) {
        HashSet<Integer> set = new HashSet<>();
        while (n != 1) {
            n = sum(n);
            if (set.contains(n))
                return false;
            else
                set.add(n);
        }
        return true;
    }
}